src/CFamily/CMake/CompilationConfigProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO.Abstractions;
+using System.Linq;
+using Microsoft.VisualStudio;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.CFamily.CMake
+{
+    internal interface ICompilationConfigProvider
+    {
+        /// <summary>
+        /// Returns the compilation configuration for the given file,
+        /// as specified in the compilation database file for the currently active build configuration.
+        /// Returns null if there is no compilation database or if the file does not exist in the compilation database.
+        /// </summary>
+        CompilationDatabaseEntry GetConfig(string analyzedFile);
```

Comment:
Minor: `filePath` might be better than `analyzedFile`
* makes it obvious that a full path is expected
* `analyzed` is leaking information about how a particular caller is using this service into the naming. This service shouldn't know or care what the caller is doing.